hnn_core/network.py

```diff
@@ -219,47 +298,436 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
-    def _add_external_feeds(self):
-        """Legacy function, for backward compatibility with original HNN-GUI
+    def copy(self):
+        """Return a copy of the Network instance.
 
-        This function is called exactly once during initialisation
+        Returns
+        -------
+        inst : instance of Network
+            A copy of the instance.
         """
-        # params of common external feeds inputs in p_common
-        # Global number of external inputs ... automatic counting
-        # makes more sense
-        # p_unique represent ext inputs that are going to go to each cell
-        self._p_common, self._p_unique = create_pext(self.params,
-                                                     self.params['tstop'])
-        self._n_common_feeds = len(self._p_common)
-
-        # 'position' the artificial cells arbitrarily in the origin of the
-        # network grid. Non-biophysical cell placement is irrelevant
-        # However, they must be present to be included in gid_ranges!
-        origin = self.pos_dict['origin']
-
-        # COMMON FEEDS
-        self.pos_dict['common'] = [origin for i in range(self._n_common_feeds)]
-
-        # UNIQUE FEEDS
-        for key in self._p_unique.keys():
-            # create the pos_dict for all the sources
-            self.pos_dict[key] = [origin for i in range(self.n_cells)]
-
-        # Now add the names of the feeds to a list
-        self.feedname_list.append('common')
-        # grab the keys for the unique set of inputs and sort the names
-        # append them to the src list along with the number of cells
-        unique_keys = sorted(self._p_unique.keys())
-        self.feedname_list += unique_keys
+        return deepcopy(self)
 
+    def add_evoked_drive(self, name, *, mu, sigma, numspikes, location,
```

Comment:
Why do we need to allow for a variable argument list here?

Reply:
I think you're confusing this with **kwargs and *args. See: https://stackoverflow.com/questions/2965271/forced-naming-of-parameters-in-python

Reply:
Oh I see. I've never used forced parameter naming before so this is educational for me!

Reply:
MeToo ;)